src/main/java/io/jenkins/plugins/sample/SampleConfiguration.java

```diff
@@ -19,33 +22,56 @@ public static SampleConfiguration get() {
         return ExtensionList.lookupSingleton(SampleConfiguration.class);
     }
 
-    private String label;
+    private String name;
+    private String description;
 
     public SampleConfiguration() {
         // When Jenkins is restarted, load any saved configuration from disk.
         load();
     }
 
     /** @return the currently configured label, if any */
-    public String getLabel() {
-        return label;
+    public String getName() {
+        return name;
     }
+    public String getDescription(){ return description; }
 
     /**
-     * Together with {@link #getLabel}, binds to entry in {@code config.jelly}.
-     * @param label the new value of this field
+     * Together with {@link #getName}, binds to entry in {@code config.jelly}.
+     * @param name the new value of this field
      */
     @DataBoundSetter
-    public void setLabel(String label) {
-        this.label = label;
+    public void setName(String name) {
+        if(name.length()!=0 && nameFormatCheck(name)) {
+            this.name = name;
+            save();
+        }
+    }
+
+    @DataBoundSetter
+    public void setDescription(String description) {
+        this.description = description;
         save();
     }
 
-    public FormValidation doCheckLabel(@QueryParameter String value) {
+    public FormValidation doCheckName(@QueryParameter String value) {
+
+        //Empty name check
         if (StringUtils.isEmpty(value)) {
-            return FormValidation.warning("Please specify a label.");
+            return FormValidation.warning("Please specify a name.");
+        }else{
+            //Check format of name (letters and hyphens only)
+            if(!nameFormatCheck(value)){
+                return FormValidation.warning("Please enter a valid name (letters and hyphens only).");
```

Comment:
but the regular expression allows whitesapce too :)
and hyphens where not part of the requirement.